flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -3,27 +3,257 @@ import Sentry
 import UIKit
 
 public class SwiftSentryFlutterPlugin: NSObject, FlutterPlugin {
-  public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "sentry_flutter", binaryMessenger: registrar.messenger())
-    let instance = SwiftSentryFlutterPlugin()
-    registrar.addMethodCallDelegate(instance, channel: channel)
-  }
-
-  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
-    result("iOS " + UIDevice.current.systemVersion)
-  }
-
-  override init() {
-    if let path = Bundle.main.path(forResource: "Info", ofType: "plist") {
-      if let resource = NSDictionary(contentsOfFile: path) {
-        if let dsn = resource.object(forKey: "SentryDsn") {
-          SentrySDK.start { options in
-              options.dsn = dsn as? String
-              options.debug = resource["SentryDebug"] as? Bool ?? false
-              options.attachStacktrace = true
-          }
-        }
-      }
-    }
-  }
+
+    var sentryOptions: Options?
+
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        let channel = FlutterMethodChannel(name: "sentry_flutter", binaryMessenger: registrar.messenger())
+        let instance = SwiftSentryFlutterPlugin()
+        registrar.addMethodCallDelegate(instance, channel: channel)
+    }
+
+    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        switch call.method as String{
+            case "deviceInfos" :
+              deviceInfos(result: result)
+              break
+
+            case "initNativeSdk" :
+              initNativeSdk(call, result: result)
+              break
+
+        case "captureEnvelope" :
+            captureEnvelope(call, result: result)
+            break
+        default:
+            result(FlutterMethodNotImplemented)
+        }
+
+
+    }
+
+    private func deviceInfos(result: @escaping FlutterResult){
+        SentrySDK.configureScope{ scope in
+            let serializedScope = scope.serialize()
+            let contexts = serializedScope["context"]
+
+            var infos = ["contexts":contexts]
+            print("contexts \(String(describing: contexts))")
+            // TODO DEBUG context
+
+            if let integrations = self.sentryOptions?.integrations {
+                infos["integrations"] = integrations
+            }
+
+            // TODO add sdk.packages
```

Comment:
should this be here? if iOS has no packages, seems irrelevant to return to the Flutter bits

Reply:
oh btw you need to add the `iOS` package to the Flutter events, so you need to have at least one package made out of the `sdk`, so it makes sense to return

`val package = Packake(sdk.name, sdk.version)`

Reply:
this info should be read from SentryMeta which is not public. I'll try to find an other way.

Reply:
not sure if I follow the problem, the naming is hardcoded and the version is available thru `options.sdk.version`

Reply:
unfortunately, neither the `sdk` or the `sdkVersion` are available in `options`
sentry-cocoa seems to add the `sdk` to the event with info from the SentryMeta static fields, and it's seems not available from anywhere else https://github.com/getsentry/sentry-cocoa/blob/c34b9067584b52e4136e5fddb7050ee6f3ca022a/Sources/Sentry/SentryClient.m#L330-L337

Reply:
✅  for now I send hardcoded value for both version and name